frontend/src/lib/experimental/NPSPrompt.tsx

```diff
@@ -0,0 +1,191 @@
+import { Button, Input } from 'antd'
+import { kea, useActions, useValues } from 'kea'
+import { FEATURE_FLAGS } from 'lib/constants'
+import { featureFlagLogic } from 'lib/logic/featureFlagLogic'
+import { CloseOutlined, ArrowLeftOutlined } from '@ant-design/icons'
+import React from 'react'
+import './NPSPrompt.scss'
+import { npsLogicType } from './NPSPromptType'
+import posthog from 'posthog-js'
+import nps from './nps.svg'
+
+const NPS_APPEAR_TIMEOUT = 10000
+const NPS_HIDE_TIMEOUT = 3500
+const NPS_LOCALSTORAGE_KEY = 'experimental-nps'
+
+type Step = 0 | 1 | 2 | 3
+
+interface NPSPayload {
+    score?: 1 | 3 | 5 // 1 = not disappointed; 3 = somewhat disappointed; 5 = very disappointed
+    feedback_score?: string
+    feedback_persona?: string
+}
+
+const npsLogic = kea<npsLogicType<NPSPayload, Step>>({
+    selectors: {
+        featureFlagEnabled: [
+            () => [featureFlagLogic.selectors.featureFlags],
+            (featureFlags) => featureFlags[FEATURE_FLAGS.NPS_PROMPT],
+        ],
+    },
+    actions: {
+        show: true,
+        hide: true,
+        setStep: (step: Step) => ({ step }),
+        stepBack: true,
+        setPayload: (payload: NPSPayload | null) => ({ payload }),
+        submit: true,
+    },
+    reducers: {
+        step: [
+            0 as Step,
+            {
+                setStep: (_, { step }) => step,
+                stepBack: (state) => Math.max(state - 1, 0) as Step,
+                submit: () => 3,
+                // go to step 1 when selecting the score on step 0
+                setPayload: (state, { payload }) => (state === 0 && typeof payload?.score !== 'undefined' ? 1 : state),
+            },
+        ],
+        hidden: [true, { show: () => false, hide: () => true }],
+        payload: [
+            null as NPSPayload | null,
+            {
+                setPayload: (state, { payload }) => ({ ...state, ...payload }),
+            },
+        ],
+    },
+    listeners: ({ values, actions, cache }) => ({
+        stepBack: () => {
+            if (values.step === 1) {
+                actions.setPayload(null)
+            }
+        },
+        submit: () => {
+            // `nps_2106` is used to identify users who have replied to the NPS survey (via cohorts)
+            const result = ['dismissed', 'partial', 'completed'][values.step]
```

Comment:
Minor bug here: we have 4 steps, might need to repeat` 'partial'` or handle steps 1,2 differently.

Reply:
ah, yes :)